pkg/asset/manifests/topologies.go

```diff
@@ -33,7 +34,7 @@ func determineTopologies(installConfig *types.InstallConfig) (controlPlaneTopolo
 		// Two node deployments with 0 workers mean that the control plane nodes are treated as workers
 		// in that situation we have decided that it is appropriate to set the infrastructureTopology to HA.
 		// All other configuration for different worker count are respected with the original intention.
-		if controlPlaneReplicas == 2 {
+		if controlPlaneTopology == configv1.DualReplicaTopologyMode || controlPlaneTopology == configv1.HighlyAvailableArbiterMode {
```

Comment:
Does this list also include controlPlaneTopology == configv1.HighlyAvailableTopologyMode ?